apps/src/componentLibrary/tabs/_Tab.tsx

```diff
@@ -0,0 +1,97 @@
+import classNames from 'classnames';
+import React, {useCallback} from 'react';
+
+import FontAwesomeV6Icon, {
+  FontAwesomeV6IconProps,
+} from '@cdo/apps/componentLibrary/fontAwesomeV6Icon';
+import moduleStyles from '@cdo/apps/componentLibrary/tabs/tabs.module.scss';
+
+export interface TabModel {
+  /** Unique value of the tab */
+  value: string;
+  /** Tab text */
+  text?: string;
+  /** Icon left from label */
+  iconLeft?: FontAwesomeV6IconProps;
+  /** Icon right from label */
+  iconRight?: FontAwesomeV6IconProps;
+  /** Whether button should be icon only */
+  isIconOnly?: boolean;
+  /** Tab icon */
+  icon?: FontAwesomeV6IconProps;
```

Comment:
Not to go back in circles but I'm thinking if there's a way through TS or otherwise we could share this behavior with the `Button` component, because the logic is very similar (can only have icon or text, and if text then iconLeft/Right is valid, otherwise not)...maybe some internal `<IconTextField/>` that handles the props and throws errors internally? Definitely not a requirement for this PR, but just calling it out as it came to mind